pkg/serviceaccounttoken/secret.go

```diff
@@ -122,3 +125,15 @@ func ServiceAccountSecretName(sa *v1.ServiceAccount) string {
 	}
 	return sa.Secrets[0].Name
 }
+
+func isSecretForServiceAccount(secret *v1.Secret, sa *v1.ServiceAccount) bool {
+	if secret.Type != v1.SecretTypeServiceAccountToken {
+		return false
+	}
+	annotations := secret.Annotations
+	if annotations == nil {
```

Comment:
Extremely minor: you don't really need this nil check. If the map is nil, you'll just get an empty string, which wouldn't be equal to SA's name, since names are always required.